### PR TITLE
Added isort for import ordering

### PR DESCRIPTION
**Pull request by fdamken**
order

### CODE REVIEW

**Comment by famura** [comment]
Honestly, I don't like the way isort sorts.
image

**Comment by famura** [comment]
Plus, I think this is incompatible with black.

**Comment by fdamken** [comment]
> Honestly, I don't like the way isort sorts.
> image

It's highly configurable and I did not expect you to like it on the first try.

**Comment by fdamken** [comment]
> Plus, the result is incompatible with black.

It can be made compatible, just a configuration thing, see e.g. https://pycqa.github.io/isort/docs/configuration/black_compatibility/

**Comment by famura** [comment]
Then we should make it compatible with black **and** match the existing style.

**Comment by miterion** [comment]
> Honestly, I don't like the way isort sorts.
> image

The suggested order is the one suggested by PEP-8 and kind of the industry standard in python: First standard lib, then external libraries, than internal stuff.

Are you sure you want to deviate from this?

**Comment by famura** [comment]
You are right, the order is fine. I guess, I was just turned off by the black incompatibility and the extra line.

**Comment by famura** [comment]
Please also run isort once for the Pyrado folder and the setup_deps.py file and the RcsPySim/setup.py file and verify that it works with Black

**Comment by famura** [comment]
This way, we have it all good and it passes its workflow test

**Comment by fdamken** [comment]
> Please also run isort once for the Pyrado folder and the setup_deps.py file and the RcsPySim/setup.py file and verify that it works with Black

I found the issue with Black compatibility, isort does not pick up the config file from the parent directory, I'll fix that,

**Comment by famura** [comment]
> > Please also run isort once for the Pyrado folder and the setup_deps.py file and the RcsPySim/setup.py file and verify that it works with Black
> 
> I found the issue with Black compatibility, isort does not pick up the config file from the parent directory, I'll fix that,

Good. I also changed the order in which they run. That should solve it the easy way xD

**Comment by fdamken** [comment]
> > > Please also run isort once for the Pyrado folder and the setup_deps.py file and the RcsPySim/setup.py file and verify that it works with Black
> > 
> > 
> > I found the issue with Black compatibility, isort does not pick up the config file from the parent directory, I'll fix that,
> 
> Good. I also changed the order in which they run. That should solve it the easy way xD

There's a small line between easy and hacky :P

**Comment by famura** [comment]
Please also add isort to the pip install command in the docker file. Then we can remove the `pip install isort` from the github action

**Comment by famura** [comment]
> > > > Please also run isort once for the Pyrado folder and the setup_deps.py file and the RcsPySim/setup.py file and verify that it works with Black
> > > 
> > > 
> > > I found the issue with Black compatibility, isort does not pick up the config file from the parent directory, I'll fix that,
> > 
> > 
> > Good. I also changed the order in which they run. That should solve it the easy way xD
> 
> There's a small line between easy and hacky :P

Happy to quote you on that one in the future :)
I guess it won't hurt if we do both solutions.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -7,13 +7,16 @@ jobs:
     name: Checking for code conformity with Black and isort
     runs-on: ubuntu-latest
     env:
-      working-directory: ./Pyrado
+      # Do not directly run in ./Pyrado as isort will not pick up the config file correctly.
+      working-directory: .
```
Comment (famura) [inline .github/workflows/lint.yml:11]: well, then we don't need the whole `env` statement, right?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -7,13 +7,16 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
       - name: Install dependencies
         run: python -m pip install --upgrade isort
-      - name: Lint isort
-        run: isort --check --diff .
       - name: Lint black
         uses: psf/black@stable
+        with:
```
Comment (famura) [inline .github/workflows/lint.yml:19]: lol, please change the order back.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -7,13 +7,16 @@ jobs:
...
     runs-on: ubuntu-latest
     env:
-      working-directory: ./Pyrado
+      # Do not directly run in ./Pyrado as isort will not pick up the config file correctly.
+      working-directory: .
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
       - name: Install dependencies
         run: python -m pip install --upgrade isort
```
Comment (famura) [inline .github/workflows/lint.yml:16]: eventually remove this line

**Review by famura — CHANGES_REQUESTED** [review]

**Comment by famura** [comment]
> docker

Ok

**Comment by famura** [comment]
Sorry. Just the change in the Dockerfile, and the initial run are missing.

**Comment by fdamken** [comment]
isort is now compatible with black, shall I push the reordered code?

**Comment by famura** [comment]
I have an additional request. Could you please separate the three linting github actions into three different actions, such that they are ran independently? That would be great.

**Review by famura — APPROVED** [review]

**Pull request merged**